backend/src/schema/resolvers/registration.spec.js

```diff
@@ -0,0 +1,402 @@
+import { GraphQLClient } from 'graphql-request'
+import Factory from '../../seed/factories'
+import { host, login } from '../../jest/helpers'
+import { neode } from '../../bootstrap/neo4j'
+
+let factory
+let client
+let variables
+let action
+let userParams
+const instance = neode()
+
+beforeEach(async () => {
+  variables = {}
+  factory = Factory()
+})
+
+afterEach(async () => {
+  await factory.cleanDatabase()
+})
+
+describe('CreateInvitationCode', () => {
+  const mutation = `mutation { CreateInvitationCode { token } }`
+
+  it('throws Authorization error', async () => {
+    const client = new GraphQLClient(host)
+    await expect(client.request(mutation)).rejects.toThrow('Not Authorised!')
+  })
+
+  describe('authenticated', () => {
+    beforeEach(async () => {
+      userParams = {
+        id: 'i123',
+        name: 'Inviter',
+        email: 'inviter@example.org',
+        password: '1234',
+      }
+      action = async () => {
+        const factory = Factory()
+        await factory.create('User', userParams)
+        const headers = await login(userParams)
+        client = new GraphQLClient(host, { headers })
+        return client.request(mutation)
+      }
+    })
+
+    it('resolves', async () => {
+      await expect(action()).resolves.toEqual({
+        CreateInvitationCode: { token: expect.any(String) },
+      })
+    })
+
+    it('creates an InvitationCode with a `createdAt` attribute', async () => {
+      await action()
+      const codes = await instance.all('InvitationCode')
+      const invitation = await codes.first().toJson()
+      expect(invitation.createdAt).toBeTruthy()
+      expect(Date.parse(invitation.createdAt)).toEqual(expect.any(Number))
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-03T18:05:52Z" title="Wednesday, July 3rd 2019, 8:05:52 pm +02:00">Jul 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,395 @@
+import { GraphQLClient } from 'graphql-request'
+import Factory from '../../seed/factories'
+import { host, login } from '../../jest/helpers'
+import { neode } from '../../bootstrap/neo4j'
+
+let factory
+let client
+let variables
+let action
+let userParams
+const instance = neode()
+
+beforeEach(async () => {
+  variables = {}
+  factory = Factory()
+})
+
+afterEach(async () => {
+  await factory.cleanDatabase()
+})
+
+describe('CreateInvitationCode', () => {
+  const mutation = `mutation { CreateInvitationCode { token } }`
+
+  it('throws Authorization error', async () => {
+    const client = new GraphQLClient(host)
+    await expect(client.request(mutation)).rejects.toThrow('Not Authorised!')
+  })
+
+  describe('authenticated', () => {
+    beforeEach(async () => {
+      userParams = {
+        id: 'i123',
+        name: 'Inviter',
+        email: 'inviter@example.org',
+        password: '1234',
+      }
+      action = async () => {
+        const factory = Factory()
+        await factory.create('User', userParams)
+        const headers = await login(userParams)
+        client = new GraphQLClient(host, { headers })
+        return client.request(mutation)
+      }
+    })
+
+    it('resolves', async () => {
+      await expect(action()).resolves.toEqual({
+        CreateInvitationCode: { token: expect.any(String) },
+      })
+    })
+
+    it('creates an InvitationCode with a `createdAt` attribute', async () => {
+      await action()
+      const codes = await instance.all('InvitationCode')
+      const invitation = await codes.first().toJson()
+      expect(invitation.createdAt).toBeTruthy()
+      expect(Date.parse(invitation.createdAt)).toEqual(expect.any(Number))
```


why do we need two expectations here?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-07-03T20:08:01Z" title="Wednesday, July 3rd 2019, 10:08:01 pm +02:00">Jul 3, 2019</time>_

---


I just want to make sure that the string returned from the backend is a valid ISO datestring. Apparently the way to check it is to parse it.

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-03T20:18:40Z" title="Wednesday, July 3rd 2019, 10:18:40 pm +02:00">Jul 3, 2019</time>_

---


sure, but then can we remove the `toBeTruthy()` expectation?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-07-03T20:55:07Z" title="Wednesday, July 3rd 2019, 10:55:07 pm +02:00">Jul 3, 2019</time>_

---


no, because `Date.parse(null)` returns a valid date

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-07-03T20:55:38Z" title="Wednesday, July 3rd 2019, 10:55:38 pm +02:00">Jul 3, 2019</time>_

---


oh, no, it's `NaN`

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-07-03T20:55:50Z" title="Wednesday, July 3rd 2019, 10:55:50 pm +02:00">Jul 3, 2019</time>_

---


new Date(null) is a valid date though

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-07-03T20:56:17Z" title="Wednesday, July 3rd 2019, 10:56:17 pm +02:00">Jul 3, 2019</time>_

---


anyhow, I think for safety I keep it there, OK?

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-03T21:29:40Z" title="Wednesday, July 3rd 2019, 11:29:40 pm +02:00">Jul 3, 2019</time>_

---


of course!
I know that you avoid multiple expectations in the same it block, just curious why you went against your normal workflow. thanks for the explanation